From d6f59d686557b645e026b023e079abb23384fe83 Mon Sep 17 00:00:00 2001
From: Nicolin Chen <b42378@freescale.com>
Date: Thu, 12 Sep 2013 14:24:20 +0800
Subject: [PATCH 0674/1543] ENGR00279368-1 ASoC: fsl: Drop useless resume
 function in WM8962 machine driver

commit da366be90891e559ef2def4909fefe0ad43867ed from
git://git.freescale.com/imx/linux-2.6-imx.git

The resume function is useless for the driver because registered jack pin has
already handled suspend/resume cases, thus drop it.

Acked-by: Wang Shengjiu <b02247@freescale.com>
Signed-off-by: Nicolin Chen <b42378@freescale.com>
(cherry picked from commit d9357cae4492abd5bca41515793ab6fe461717b2)
---
 sound/soc/fsl/imx-wm8962.c |   46 --------------------------------------------
 1 files changed, 0 insertions(+), 46 deletions(-)

diff --git a/sound/soc/fsl/imx-wm8962.c b/sound/soc/fsl/imx-wm8962.c
index 565e387..6e9d3d3 100644
--- a/sound/soc/fsl/imx-wm8962.c
+++ b/sound/soc/fsl/imx-wm8962.c
@@ -165,41 +165,6 @@ static int micjack_status_check(void)
 	return ret;
 }
 
-static void imx_resume_event(struct work_struct *wor)
-{
-	struct imx_priv *priv = &card_priv;
-	struct snd_soc_jack *jack;
-	int enable, report;
-
-	if (gpio_is_valid(priv->hp_gpio)) {
-		jack = imx_hp_jack_gpio.jack;
-
-		enable = gpio_get_value_cansleep(imx_hp_jack_gpio.gpio);
-		if (imx_hp_jack_gpio.invert)
-			enable = !enable;
-
-		report = enable ? imx_hp_jack_gpio.report : 0;
-
-		snd_soc_jack_report(jack, report, imx_hp_jack_gpio.report);
-	}
-
-	if (gpio_is_valid(priv->mic_gpio)) {
-		jack = imx_mic_jack_gpio.jack;
-
-		enable = gpio_get_value_cansleep(imx_mic_jack_gpio.gpio);
-		if (imx_mic_jack_gpio.invert)
-			enable = !enable;
-
-		report = enable ? imx_mic_jack_gpio.report : 0;
-
-		snd_soc_jack_report(jack, report, imx_mic_jack_gpio.report);
-	}
-
-	return;
-}
-
-static DECLARE_DELAYED_WORK(resume_jack_event, imx_resume_event);
-
 
 static const struct snd_soc_dapm_widget imx_wm8962_dapm_widgets[] = {
 	SND_SOC_DAPM_HP("Headphone Jack", NULL),
@@ -398,16 +363,6 @@ static ssize_t show_mic(struct device_driver *dev, char *buf)
 
 static DRIVER_ATTR(microphone, S_IRUGO | S_IWUSR, show_mic, NULL);
 
-static int imx_wm8962_resume(struct snd_soc_card *card)
-{
-	struct imx_priv *priv = &card_priv;
-
-	if (gpio_is_valid(priv->hp_gpio) || gpio_is_valid(priv->mic_gpio))
-		schedule_delayed_work(&resume_jack_event, msecs_to_jiffies(200));
-
-	return 0;
-}
-
 static int imx_wm8962_late_probe(struct snd_soc_card *card)
 {
 	struct snd_soc_dai *codec_dai = card->rtd[0].codec_dai;
@@ -545,7 +500,6 @@ static int imx_wm8962_probe(struct platform_device *pdev)
 	data->card.num_dapm_widgets = ARRAY_SIZE(imx_wm8962_dapm_widgets);
 
 	data->card.late_probe = imx_wm8962_late_probe;
-	data->card.resume_post = &imx_wm8962_resume;
 
 	platform_set_drvdata(pdev, &data->card);
 	snd_soc_card_set_drvdata(&data->card, data);
-- 
1.7.5.4

